be/src/exec/pipeline/schedule/observer.h

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <atomic>
+#include <vector>
+
+#include "exec/pipeline/pipeline_fwd.h"
+#include "exec/pipeline/schedule/common.h"
+#include "exec/pipeline/schedule/utils.h"
+#include "runtime/runtime_state.h"
+#include "util/defer_op.h"
+#include "util/race_detect.h"
+
+namespace starrocks::pipeline {
+class SourceOperator;
+
+// Pipeline observer objects. the Observable issues an event to the Observer.
+// thread-safe.
+// If the driver satisfies the ready state,
+// the observer adds the corresponding driver to the ready driver queue.
+// If the pipeline driver is not in the block queue, then the pipeline driver tries to tag it.
+// We need to handle this in the event scheduler.
+class PipelineObserver {
+public:
+    PipelineObserver(DriverRawPtr driver) : _driver(driver) {}
+
+    DISALLOW_COPY_AND_MOVE(PipelineObserver);
+
+    void source_update() {
```

Comment:
update is quite a common word,  can we use trigger, fire, snap or other names?

Reply:
I would use trigger instead of update.